apps/traincascade/cascadeclassifier.cpp

```diff
@@ -450,6 +452,7 @@ void CvCascadeClassifier::save( const string filename, bool baseFormat )
         CvSeq* weak;
         if ( cascadeParams.featureType != CvFeatureParams::HAAR )
             CV_Error( CV_StsBadFunc, "old file format is used for Haar-like features only");
+        fs << "{:" << ICV_HAAR_TYPE_ID;
```

Comment:
I checked it one more time. `fs << "{:" << ICV_HAAR_TYPE_ID;` is probably not what you want. You'd want something like `fs << "{:" ICV_HAAR_TYPE_ID;` to make sure that `"{:opencv-haar-classifier"` is streamed to fs and parsed as a single "token"